yamlpatch/patch_test.go

```diff
@@ -94,6 +94,50 @@ foo:
 foo:
   arr: [0, 1, 2, 3]
   bar: 1 # my bar`,
+		},
+		{
+			Name:  "add to array with comment",
+			Patch: []string{`{"op":"add","path":"/foo/arr/-","value":4,"comment":"the number 4"}`},
+			Body: `# my foo
+foo:
+  arr:
+    # numbers 1 through 3
+    - 1
+    - 2
+    - 3`,
+			Expected: `# my foo
+foo:
+  arr:
+    # numbers 1 through 3
+    - 1
+    - 2
+    - 3
+    # the number 4
+    - 4`,
+		},
+		{
+			Name:  "replace object with comment",
+			Patch: []string{`{"op":"replace","path":"/foo/bar","value":{"key":"value"},"comment":"key value pair"}`},
+			Body: `# my foo
+foo:
+  bar: hello-world # previous comment`,
+			Expected: `# my foo
+foo:
+  bar:
+    # key value pair
+    key: value`,
+		},
+		{
+			Name:  "test object with comment",
```

Comment:
looks like this is the same test as above?

Reply:
Whoops, fixed